specs/interop.md

```diff
@@ -0,0 +1,324 @@
+
+The execution payload MUST set `noTxPool` to `false` for this block.
+
+The exact definitions for these upgrade transactions are still to be defined.
+
+## Derivation Pipeline
+
+The derivation pipeline enforces invariants on safe blocks that include relaying transactions.
+
+- The relaying message MUST have a corresponding initiating message
+- The initiating message that corresponds to a relaying message MUST come from an allowed chain (dependency set)
+
+Blocks that contain transactions that relay cross domain messages to the remote chain where the
+initiating transaction does not exist MUST be considered invalid and MUST not be allowed by the
+derivation pipeline to be considered safe.
+
+There is no concept of replay protection within the protocol since there is no replay protection
+mechanism that fits well for all applications. Users MAY submit an arbitrary number of relaying
+messages per initiating message. This greatly simplifies the protocol and allows for
+applications to build application specific replay protection.
+
+### Depositing a Relaying Message
+
+Deposit transactions (force inclusion transactions) give censorship resistance to layer two networks.
+The derivation pipeline must gracefully handle the case in which a user uses a deposit transaction to
+relay a cross chain message. To not couple preconfirmation security to consensus, deposit transactions
+that relay cross chain messages MUST have an initiating message that is considered safe. This relaxes
+a strict synchrony assumption on the sequencer that it MUST have all unsafe blocks of remote chains
+as fast as possible to ensure that it is building correct blocks. It also prevents a class of attacks
+where the user can send a relaying message before the initiating message and trick the derivation pipeline
+into reorganizing the sequencer.
+
+### Safety
+
+The initiating messages for all relaying messages MUST be resolved as safe before an L2 block can transition from being unsafe to safe.
+Users MAY optimistically accept unsafe blocks without any verification of the relaying messages. They SHOULD optimistically verify
+the initiating messages exist in remote unsafe blocks to more quickly reorganize out invalid blocks.
+
+## State Transition Function
+
+After the full execution of a block, the set of `MessagePassed` events emitted from the `CrossL2Outbox` are collected and merklized.
+This commitment MUST be included as the block header's [extra data field](https://github.com/ethereum/execution-specs/blob/1fed0c0074f9d6aab3861057e1924411948dc50b/src/ethereum/frontier/fork_types.py#L115).
+The events are serialized with using [Simple Serialize](https://github.com/ethereum/consensus-specs/blob/dev/ssz/simple-serialize.md) aka SSZ.
+
+| Name          | Value |
+|---------------|-------------------------------------|
+| `MAX_LOG_DATA_SIZE` | `uint64(2**24)` or 16,777,216 |
+| `MAX_MESSAGES_PER_BLOCK` | `uint64(2**20)` or 1,048,576 |
+
```

Comment:
I decided for ssz here because I am under the impression that it is better for the fault proof. I don't love how there are 2 consensus serializations of the cross chain message. One is ABI for the smart contracts (see the `messageHash` definition) and then ssz for in the extradata. Thoughts @protolambda?

Reply:
Yeah SSZ is definitely better here to avoid the large preimage problem. The SSZ Merkle Tree breaks everything down into a maximum leaf size of 32 bytes whereas ABI format would represent the entire event data as a single preimage.  If the event data has a size limit that would be fine, but if it has arbitrary size (or could be quite large) then we should use SSZ.